packages/core/src/components/tree/treeNode.tsx

```diff
@@ -71,7 +78,15 @@ export interface ITreeNodeProps extends ITreeNode {
     path: number[];
 }
 
-export class TreeNode extends React.Component<ITreeNodeProps, {}> {
+export class TreeNode<T = {}> extends React.Component<ITreeNodeProps<T>, {}> {
+    public static ofType<T>() {
+        return TreeNode as new () => TreeNode<T>;
+    }
+
+    constructor(props?: ITreeNodeProps<T>, context?: any) {
```

Comment:
remove

Reply:
The code won't compiled without, because of the following function:
public static ofType<T>() {
        return TreeNode as new () => TreeNode<T>;
 }

It is necessary....

Reply:
oh snap! yeah it does fail. but that's just a typings issue. 

`props?` in constructor is actually wrong--it goes against React's types and will fail if `strictNullChecks` is enabled (it is disabled in blueprint).

i'll resolve that in all instances later 👍